tidy3d/plugins/near2far/near2far.py

```diff
@@ -158,11 +421,11 @@ def fields_cartesian(self, x, y, z):
         Parameters
         ----------
         x : float
-            (micron) x distance from center of monitor.
+            (micron) x position relative to monitor center.
```

Comment:
so is it relative to monitor center still? or the `origin` of the `Near2Far`?

Reply:
Should be relative to the `origin` of `Near2Far` - maybe I can just say "relative to the centroid of all surface monitors" to be more explicit

Reply:
maybe instead of "centroid", we should say "average of the centers" if that's how it's computed?  I for one might not know what a centroid is in this case without looking it up.   I'm also wondering whether there are some pitfalls with doing it this way.  For example, the origin of the 5 sided box monitors (missing "z-" bottom) will end up slightly higher than the center of the volume enclosed by the boxes, potentially confusing people. @momchil-flex thoughts?

Reply:
Basically the question is what should be the "origin" for a collection of near2far surfaces.  possible options:
* Simulation.center
* (0,0,0) in global coordinates
* average of the surface monitor centers.
* Pick your own, or pick a monitor to use as the center (perhaps as a special option)
* other ideas?

Reply:
The origin is used not only to define observation points, but also in the near2far approximation. Some of the steps in the math use the local origin as a proxy for all source points. So even if we use Simulation.center to define observation coordinates, we'll still need a local origin to make the far field approximations. For the latter, the small shifts due to missing monitors shouldn't really matter, since any such change should be much smaller than the distance to the observation points.

Reply:
Hm.  So the local origin should probably be the individual monitor centers?  (as opposed to the average?). Unless I'm not understanding?  And then the origin to define the observation points could just be global origin (0,0,0) unless otherwise specified?  What is the scheme now?

Reply:
Right now, the local origin is the average of all monitors' centers, for the purposes of both computing the phase terms and defining the observation points. For the phase terms, yes, we could use the individual monitors' center - that's different from what Taflove does, and I'm not sure how it will impact the accuracy